internal/impl/cache/receiver_cache_request_impl_test.go

```diff
@@ -0,0 +1,249 @@
+// pubsubplus-go-client
+//
+// Copyright 2021-2024 Solace Corporation. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"math"
+	"testing"
+
+	"solace.dev/go/messaging/pkg/solace/cache"
+	"solace.dev/go/messaging/pkg/solace/resource"
+)
+
+func TestNewCachedMessageSubscriptionRequestWithInvalidPositiveCachedMessageSubscriptionStrategy(t *testing.T) {
+	cacheRequest := NewReceiverCacheRequestImpl()
+	const cacheName = "test-cache"
+	const subscriptionString = "some-subscription"
+	subscription := resource.TopicSubscriptionOf(subscriptionString)
+	const invalidCachedMessageSubscriptionStrategy = 5 // invalid subscription strategy
+	const cacheAccessTimeout = 3001                    // in valid range - (valid range from 3000 to signed MaxInt32)
+	const maxCachedMessages = 5                        // in valid range (valid range from 0 to signed MaxInt32)
+	const cachedMessageAge = 15                        // in valid range (valid range from 0 to signed MaxInt32)
+	cachedMessageSubscriptionRequest, err := cacheRequest.NewCachedMessageSubscriptionRequest(invalidCachedMessageSubscriptionStrategy,
+		cacheName, subscription, cacheAccessTimeout, maxCachedMessages, cachedMessageAge)
+
+	if cachedMessageSubscriptionRequest != nil || err == nil {
+		t.Error("Expected error while calling NewCachedMessageSubscriptionRequest() with invalid cachedMessageSubscriptionStrategy (5)")
+	}
+}
+
+func TestNewCachedMessageSubscriptionRequestWithInvalidNegativeCachedMessageSubscriptionStrategy(t *testing.T) {
+	cacheRequest := NewReceiverCacheRequestImpl()
+	const cacheName = "test-cache"
```

Comment:
I guess the way these tests are embedded into production code prevents us from factoring out the common parts? 
My uninformed googling says if the copy-pasting gets out of hand, we can try "table driven tests" at the cost of readability  :-/
https://go.dev/wiki/TableDrivenTests

Reply:
@gszol Yes, good suggestion 👍 . I can refactor the tests to use tables so we keep the test code DRY.

Reply:
Can this conversation be resolved @gszol , or is action still necessary?